### PR TITLE
Remove support of Canvas Layers on iOS

### PR DESCRIPTION
**Pull request by ASalavei**
Fixes https://youtrack.jetbrains.com/issue/CMP-6600/Remove-support-of-CanvasLayers-on-iOS

### Breaking Changes - iOS
- Remove obsolete Canvas Layers mode on iOS

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt [hunk 1]
```diff
@@ -309,17 +302,15 @@ internal class ComposeSceneMediator(
...
-            view = viewForKeyboardOffsetTransform,
+            view = view,
             keyboardOverlapHeightChanged = { height ->
-                if (configuration.platformLayers) {
-                    if (keyboardOverlapHeight != height) {
-                        animateKeyboardOffsetChanges = false
-                    }
+                if (keyboardOverlapHeight != height) {
+                    animateKeyboardOffsetChanges = false
+                    keyboardOverlapHeight = height
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt:312]: Previously it wasn't under the condition. Looks suspicious. Let's avoid introducing unrelated changes.

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt [hunk 2]
```diff
@@ -345,7 +336,7 @@ internal class ComposeSceneMediator(
     val hasInvalidations: Boolean
         get() = scene.hasInvalidations() || keyboardManager.isAnimating
 
-    private fun hitTestInteropView(point: CValue<CGPoint>, event: UIEvent?): UIView? =
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt]: Please revert unrelated changes

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt (hunk 1 above)
Comment (ASalavei) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt:312]: It is code optimisation. Should be like this.

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt (hunk 2 above)
Comment (ASalavei) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt]: That was an attempt to fix warnings in the file. Reverted.

**Review by MatkovIvan — APPROVED** [review]

**Review by mazunin-v-jb — APPROVED** [review]

**Comment by igordmn** [comment]
Changed to "Breaking Changes"
